### ISSUE TITLE
Recent support for ESP-IDF 5.0 broke 4.4

### ISSUE THREAD

**Issue by cwoolum** [pulse-loop/bluedroid#17]
After getting the latest code from main, I now receive the error

```
error[E0425]: cannot find value `DUPL_SCAN_CACHE_REFRESH_PERIOD` in this scope
   --> bluedroid\src\gatt_server\mod.rs:277:38
    |
277 |             dup_list_refresh_period: DUPL_SCAN_CACHE_REFRESH_PERIOD as u16,
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0624]: method `set_raw` is private
   --> bluedroid\src\gatt_server\custom_attributes.rs:99:22
    |
99  |                     .set_raw(&key, &value)
    |                      ^^^^^^^ private method
    |
   ::: C:\Users\machine\.cargo\registry\src\index.crates.io-6f17d22bba15001f\esp-idf-svc-0.45.0\src\nvs.rs:329:5
    |
329 |     fn set_raw(&mut self, name: &str, buf: &[u8]) -> Result<bool, EspError> {
    |     ----------------------------------------------------------------------- private method defined here
```

In `src/gatt_server/mod.rs#277`, it looks like there may have been intent to remove that field if < IDF 5.0 but it got commented out. Is this a bug or was it intentional?

```
// #[cfg(any(esp_idf_version = "5.0", esp_idf_version = "5.1"))]
dup_list_refresh_period: DUPL_SCAN_CACHE_REFRESH_PERIOD as u16,
```

I'm using ESP-IDF 4.4 and using ESP32-C3.

**Comment by persello** [comment]
I'll have a look at it. It's strange that the CI passed, because it builds for both 4.4 and 5.0.

By chance, do you use an intermediate version of IDF, between 4.4 and 5.0? I had an issue with v4.4.2 when adding the new CI tests.

**Comment by Vollbrecht** [comment]
site note: i see that you are using release/4.4 and release/5.0 in your ci build. You have to be aware that this are not fixed releases, rather moving git targets. If you want to go for stability or repeatability, you should build against the release tags aka v4.4.4 or v5.0.1 or you wake up and the next day nothing goes :smile: Keep up the good work !

**Comment by persello** [comment]
For some reason, when I wrote the Actions, I thought it was the opposite case.

**Comment by Vollbrecht** [comment]
> 

yeah, first time i also was confused there. you can also build for both, so you get alarmed when upstream breaks in future. Best of both worlds.

**Comment by cwoolum** [comment]
I am using `"release/v4.4"`.

**Comment by persello** [comment]
That might be a rolling release issue. I might solve it in this repository, although I'd prefer to fix it in esp-idf-svc, as this repository is being merged there.

**Comment by cwoolum** [comment]
I have PR #18 that resolves the issues I was running into.

**Comment by cwoolum** [comment]
Let me know if I should PR there directly.

**Comment by cwoolum** [comment]
After running into a few more issues, I discovered the problem. It seems that new bluetooth fields were released in `4.4.4`. I've tried updating some of the conditional build flags to specifically cover that.

```
#[cfg(any(
       esp_idf_version = "4.4.4",
       esp_idf_version = "5.0",
       esp_idf_version = "5.1"
))]
```

~~The problem is that `esp_idf_version` doesn't seem to be that granular and builds for ESP-IDF `4.4.4` fail because the respective fields are not being compiled. Are you aware of a workaround for this?~~

*EDIT:* NM, figured out how to target specific patch versions

You can see the failing builds [here](https://github.com/cwoolum/bluedroid/actions/runs/4692682367)

**Comment by persello** [comment]
I merged your PR to develop. Thanks for your fix.
